backend/routes/auth.js

```diff
@@ -13,26 +13,44 @@ const { default: mongoose } = require("mongoose");
  * @param {string} req.body.email - The user's email.
  * @param {string} req.body.password - The user's password.
  */
-router.post("/login", async (req, res) => {
-  console.log("Request Body: ", req.body);
-  const { error } = validateLogin(req.body);
-  if (error) return res.status(400).send(error.details[0].message);
-
-  let user = await User.findOne({ email: req.body.email });
-  if (!user)
-    return res.send(getErrorResponse("No User Exists with this email"));
 
-  const validPassword = req.body.password === user.password;
-  if (!validPassword) return res.send(getErrorResponse("Invalid Password"));
 
-  return res.send(
-    getSuccessResponse(
-      "Login Success",
-      _.omit(user.toObject(), ["password", "__v"])
-    )
-  );
+router.post("/login", async (req, res) => {
+  try {
+    console.log("Request Body: ", req.body);
+
+    // Validate the login request(🚦)
+    const { error } = validateLogin(req.body);
+    if (error) {
+      return res.status(400).send(getErrorResponse("Invalid request data"));
```

Comment:
Actually, I want the following here so that React doesn't throw errors. 

```js
return res.send(getErrorResponse("Invalid request data"));
```

The use of `.status(400)` causes the front end to think that an error has occurred. We can do two things here:

1. Use the approach you are using, but then we have to catch all the errors in the front end. (and we will have to do this for every endpoint).
2. Use the previously used approach and simply get the data as expected, in the frontend. As we are using this throughout the endpoints, this must be easy.

Let me know which approach will be better.

Reply:
i will follow your approach that means i will just have to use single status code to catch all the errors.